chore: add 1px border to Notification `SECONDARY`

fk opened the pull request:
- [x] add `theme.borderRadius[2]`
- [x] add border to variant `SECONDARY` via `box-shadow` to make them a little more background-agnostic/stand out on light-but-not-white backgrounds
- [x] verify that adding `theme.borderRadius[2]` doesn't cause any regressions in Gatsby Cloud (the `SECONDARY` variant originally didn't have a `borderRadius` set, so was ready to be used as "full width" component)

Extracted from #271.

Following up on this, let's discuss adding an option to disable `border-radius` to support using the component as full-width "banner" (see [Figma spec](https://www.figma.com/file/h4ixUmOo781r3sDeBAbmDc/Notifications?node-id=0%3A1))

Elanhant commented:
@fk I've performed an audit of current "SECONDARY" `Notification` variants in Cloud, everything looks good to me:
### Error Alert
![localhost_8000_dashboard_organization_cf77a3b7-b467-49b8-bab7-c3298289a7a0_members (5)](https://user-images.githubusercontent.com/4366711/96305116-33a55280-0fb2-11eb-9ebb-e6b544b3eef8.png)

### Importing a site
![localhost_8000_dashboard_organization_cf77a3b7-b467-49b8-bab7-c3298289a7a0_sites_import_ff556e46-2363-438e-8900-e09efa68d2f0_setup](https://user-images.githubusercontent.com/4366711/96305155-4a4ba980-0fb2-11eb-8602-07dbc08e0b30.png)

### Provisioning a site
![localhost_8000_dashboard_organization_cf77a3b7-b467-49b8-bab7-c3298289a7a0_sites_import_ff556e46-2363-438e-8900-e09efa68d2f0_setup (3)](https://user-images.githubusercontent.com/4366711/96305452-bfb77a00-0fb2-11eb-89ca-ad42ab3a18db.png)

### CDN integration (can't wait to make into a panel 😛)
(please disregard the weird lilac horizontal line, it's some glitch when taking a screenshot using Chrome dev tools) 
![localhost_8000_dashboard_organization_cf77a3b7-b467-49b8-bab7-c3298289a7a0_sites_import_ff556e46-2363-438e-8900-e09efa68d2f0_setup (4)](https://user-images.githubusercontent.com/4366711/96305318-91399f00-0fb2-11eb-9dab-f663b0b71427.png)

fk commented:
LGTM, thank you @Elanhant for reviving this PR! 🤗

I updated the PR description and moved 

> add an option to disable `border-radius` to support using the component as full-width "banner" (see [Figma spec](https://www.figma.com/file/h4ixUmOo781r3sDeBAbmDc/Notifications?node-id=0%3A1))

out of the list of todo's — will wait with merging tho! ✌️

fk commented:
Yay! 🙏 🤗
